build-tools/packages/build-cli/src/library/release.ts

```diff
@@ -159,10 +178,77 @@ export function toReportKind(
 			break;
 		}
 
+		case "legacy-compat": {
+			for (const [pkg, details] of Object.entries(report)) {
+				toReturn[pkg] = details.ranges.legacyCompat;
+			}
+			break;
+		}
+
 		default: {
 			throw new Error(`Unexpected ReportKind: ${kind}`);
 		}
 	}
 
 	return toReturn;
 }
+
+/**
+ * Generates a new version representing the next version in a legacy compatibility range based on a specified multiple of minor versions.
+ *
+ * @param version - A string representing the current version.
+ * @param interval - The multiple of minor versions to use for calculating the next version in the range.
+ * @param releaseGroup - Release group
+ * @param schemeToUse - The version scheme.
+ *
+ * @returns A string representing the next version in the legacy compatibility range.
+ */
+export function getLegacyCompatVersionRange(
+	version: string,
+	interval: LegacyCompatInterval,
+	releaseGroup: ReleaseGroup | undefined,
+	schemeToUse: VersionScheme,
+): string {
+	for (const [group, intervalValue] of Object.entries(interval.legacyCompatInterval)) {
+		if (group === releaseGroup && intervalValue > 0) {
+			return getLegacyCompatRange(version, intervalValue);
+		}
+	}
+
+	// If legacy compat range is equal to 0, return caret version.
+	return schemeToUse === "internal" ? getVersionRange(version, "^") : `^${version}`;
+}
+
+/**
+ * Generates a new version representing the next version in a legacy compatibility range for any release group.
+ *
+ * @param version - A string representing the current version.
+ * @param interval - The multiple of minor versions to use for calculating the next version.
+ *
+ * @returns A string representing the next version in the legacy compatibility range.
+ */
+export function getLegacyCompatRange(version: string, interval: number): string {
+	const semVersion = semver.parse(version);
+	if (!semVersion) {
+		throw new Error("Invalid version string");
+	}
+	// Calculate the next compatible minor version using the compatVersionInterval
+	const baseMinor = Math.floor(semVersion.minor / interval) * interval;
+	const newSemVerString = `${semVersion.major}.${baseMinor + interval}.0`;
```

Comment:
Since this does not really support prerelease (including internal) versions, there should be a check for that.
Should add a test case too.